src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -0,0 +1,726 @@
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Category('Body')]
+        [System.Collections.Hashtable]
+        # Arc Agentry System Protected Configuration
+        ${ConfigurationProtectedSetting},
+
+        [Parameter()]
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Category('body')]
+        [System.Management.Automation.SwitchParameter]
+        ${EnableGateway},
+
+        [Parameter()]
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Category('body')]
+        [System.Management.Automation.SwitchParameter]
+        ${DisableGateway},
+
+        [Parameter()]
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Category('body')]
+        [System.String]
+        # Arc Gateway resource Id
+        ${GatewayResourceId},
+        
+        [Parameter(ParameterSetName='Set', Mandatory)]
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240701Preview.IConnectedCluster]
+        ${InputObject}
+    )
+
+    process {
+        . "$PSScriptRoot/helpers/HelmHelper.ps1"
+        . "$PSScriptRoot/helpers/ConfigDPHelper.ps1"
+        . "$PSScriptRoot/helpers/AZCloudMetadataHelper.ps1"
+        if($AzureHybridBenefit){
+            if(!$AcceptEULA){
+                $legalTermPath = Join-Path $PSScriptRoot -ChildPath "LegalTerm.txt"
+                try {
+                    $legalTerm = (Get-Content -Path $legalTermPath) -join "`r`n"
+                } catch {
+                    Write-Error "Get legal term failed."
+                    throw
+                }
+                $confirmation = Read-Host $legalTerm"`n[Y] Yes  [N] No  (default is `"N`")"
+                if($confirmation -ine "Y"){
+                    Return
+                }
+            }
+        }
+        $null = $PSBoundParameters.Remove('AcceptEULA')
+
+        if ($PSBoundParameters:KubeConfig) {
+            $Null = $PSBoundParameters.Remove('KubeConfig')
+        } elseif (Test-Path Env:KUBECONFIG) {
+            $KubeConfig = Get-ChildItem -Path Env:KUBECONFIG
+        } elseif (Test-Path Env:Home) {
+            $KubeConfig = Join-Path -Path $Env:Home -ChildPath '.kube' | Join-Path -ChildPath 'config'
+        } else {
+            $KubeConfig = Join-Path -Path $Home -ChildPath '.kube' | Join-Path -ChildPath 'config'
+        }
+        if (-not (Test-Path $KubeConfig)) {
+            Write-Error 'Cannot find the kube-config. Please make sure that you have the kube-config on your machine.'
+            return
+        }
+        if ($PSBoundParameters:KubeContext) {
+            $Null = $PSBoundParameters.Remove('KubeContext')
+        }
+        if (($null -eq $KubeContext) -or ($KubeContext -eq '')) {
+            $KubeContext = kubectl config current-context
+        }
+
+        # XW TODO: unit test this?
+        # If EnableGateway is provided then set the gateway as enabled.
```

Comment:
Do we allow `GatewayResourceId` without `EnableGateway`?  If so then this should be possible:
- `New` with no gateway
- `Set` with `GatewayResourceId` - still disabled
- `Set` with `EnableGateway` - not it's enabled.

Reply:
See comments below.